components/textfield/index.css

```diff
@@ -98,7 +96,8 @@ governing permissions and limitations under the License.
   /* Use padding instead of text-indent because text-indent does not left align the text in Edge browser  */
   text-indent: 0;
 
-  inline-size: 100%;
+  min-inline-size: var(--spectrum-textfield-texticon-min-width);
```

Comment:
what if you left the size on the Textfield and instead did
```
min-inline-size: inherit;
```
?

Reply:
PS, this whole issue is a pain, we've been trying to figure out how to deal with it as well

for instance, if you have a component that is TextArea and another that is TextField, and you set height (or width) on them both, should it be the height of the entire component, including label and helptext potentially? or should it be the height of the input itself? For almost all cases, it makes sense for it to be everything, but for TextArea, this would involve a lot of math, see https://github.com/adobe/react-spectrum/pull/2515/files
Keep in mind there is also label placement side and top

And you probably don't want height and width to go to different elements

We've actually made it impossible to horizontally change the width of a textarea to cut down on this issue a little bit.

Reply:
So if we have any size at all on the outer element, then it doesn't resize to match the inner textarea when it's resized. My solution was to let the child element just dictate the size of the parent, which is fine, but then you can't set `width`/`height` on the parent element, and the input is inside of the shadow DOM in web components, so...

Reply:
Due to how we relate the `<textarea>` to the `<label>` and the `<div>` that delivers "help text" we found no workable approach to this solution that continued to leverage the `:after` pseudo element for this style and manually move the rules applied in this location onto the `<textarea>` as seen in https://github.com/adobe/spectrum-web-components/commit/fa0ac3437b269f009ef47d080b41996fe0a25779

There's definitely lots of places where this could fall down AS THE DESIGN CHANGES OVER TIME, but it seems to get both the sizing and the focus visibility correct here in a way that I couldn't recreate with any other fancy CSS approach that I tried. Particularly the idea the in SWC this needs to support both the ability to resize the `<textarea>` manually as well as the fact that we delivery a `grows` variant that resizes to the contained text, I'm not sure there's another approach that would work for us in this case.